plugins/modules/system/ssh_config.py

```diff
@@ -209,6 +209,8 @@ def ensure_state(self):
         hosts_removed = []
         hosts_added = []
 
+        hosts_result = [host for host in hosts_result if host['host'] == self.host]
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add tests for this change.

Reply:
@drenout thanks for coming up with the PR so promptly :-) as mentioned by @Akasurde you will need a changelog fragment.

Besides that, I think the problem is slightly bigger than that. We can have multiple hosts *and* negation operator on the hosts. I happen to use this in my work computer .ssh/config:
```
Host !jb-*.lab.internal *.lab.* *.lv426.* *.labx.* *.laby.*
	  ProxyJump jb-1.lab.internal
```

Reply:
Thanks @russoz 😊 Could you please elaborate more on the problem you mentioned? 
I think, if you have this .ssh/config file, then, let's say, running Ansible playbook to add host `prod.lab.com` will add a new entity, which is expected - most probably you don't want Ansible to mess up with the original line in the config. But I'm not sure about the precedence order, will the ssh reach the most specific entity for the `prod.lab.com` or will handle at according to the `!jb-*.lab.internal *.lab.* *.lv426.* *.labx.* *.laby.*` pattern

Reply:
@drenout regarding the precedence, first match wins (that's why `Host *`, if existing, must be the last one: it can provide a default value for almost any SSH option not defined in the general part of the file nor in previous host sections).

@russoz, I don't think the module is currently able to smartly manage host sections regarding their substrings and glob patterns and multi-hosts sections. For `Host foo bar barosa`, stormssh parser outputs a spec for host `'foo bar barosa'` verbatim. It is the job of the caller to add more granularity in the results, that is currently not the case, given the topic of this PR :)

Reply:
@drenout @quidame yeah, I think you are right. I thought it might be a problem and dropped the question without much analysis. At the end of the day, the users will be responsible for managing whatever they have in there.

Reply:
@russoz IMHO this module could/should handle some *simple* cases, but I guess that even the most simple would need a good serializer to be handled correctly.